primitives/trie/src/lib.rs

```diff
@@ -184,7 +184,11 @@ pub fn delta_trie_root<L: TrieConfiguration, I, A, B, DB, V>(
 	DB: hash_db::HashDB<L::Hash, trie_db::DBValue>,
 {
 	{
-		let mut trie = TrieDBMut::<L>::from_existing(&mut *db, &mut root)?;
+		let mut trie = if root == Default::default() {
```

Comment:
So we are making the default hash the root of an empty trie.
It feels a bit incorrect, the alternative would be to initiate trie backend root hash in 'new_in_mem' (maybe in other place) to the correct value of the empty trie hash.

Reply:
I realize this heuristic was already here in existing code, maybe more restrained to 'in_memory_backend.rs'

Reply:
Ahh okay, I got confused and thought this was the correct way. 

I see. I will need to find a better way.

How can I get the "correct value of the empty trie hash"?